include/pybind11/common.h

```diff
@@ -657,4 +657,32 @@ static constexpr auto const_ = std::true_type{};
 
 #endif // overload_cast
 
+NAMESPACE_BEGIN(detail)
+// Aids in converting arbitrary containers into a vector; implicitly convertible from any standard
+// container (or C-style array) supporting std::begin/std::end.
+template <typename T>
+struct any_container {
+    std::vector<T> v; // Intended to be std::move'd
```

Comment:
Nitpicking: if you don't want to make `v` public, you could `operator std::vector<int>&&() && { return std::move(v); }` and then move the vector directly out of the container.